src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -83,20 +93,52 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
             // Description
             if (displayAttribute != null)
             {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
+                if (localizer != null &&
```

Comment:
@dougbu, I'm with Taylor on this one, condensing these down into one block helps de-duplicate an already repetitive section, and would also let us trap that `localizer` inside a single if making it clear that it's only used by these.